checksit/generic.py

```diff
@@ -552,6 +554,132 @@ def check_file_name(file_name, vocab_checks=None, rule_checks=None, **kwargs):
 
     return errors, warnings
 
+def check_generic_file_name(file_name, vocab_checks=None, segregator=None, extension=None, spec_verbose=False, **kwargs):
+    # Requires yaml file containing a list of file name fields and segregators
+    # Loop over each file field and segregator until there are no more
+    # check against defined file extension
+
+    vocab_checks = vocab_checks or {}
+    try:
+        seg = segregator["seg"]
+    except:
+        seg='_'
+    try:
+        ext = extension["ext"]
+    except:
+        ext = '.test'
+    try:
+        spec_verb = spec_verbose["spec_verb"]
+    except:
+        spec_verb = False
+
+    errors = []
+    warnings = []
+
+    # get filename parts
+    if not isinstance(file_name,str):
+        raise ValueError
+
+    extracted_name = file_name.replace(ext,'')
+    file_name_parts = extracted_name.split(seg)
+
+    if spec_verb:
+        print(f"File name: {file_name}")
+        print(f"Segregator: {seg}")
+        print(f"Extension: {ext}")
+        print(f"All file name parts: {file_name_parts}")
+
+    # Loop over file name parts
+    for idx, key in enumerate(file_name_parts):
+        if spec_verb:
+            print('')
+            print(idx, key)
+        num=f"{idx:02}"
+        
+        # Check if number of file name parts matches the number of fields specified in the user-defined yaml file
+        if len(vocab_checks) < len(file_name_parts):
+            errors.append(
+                        f"[file name]: Number of file name fields ({len(file_name_parts)}) is greater than the {len(vocab_checks)} fields expected."
+                    )
+            if spec_verb:
+                print(errors[-1])
+            break
+        elif len(vocab_checks) > len(file_name_parts):
+            errors.append(
+                        f"[file name]: Number of file name fields ({len(file_name_parts)}) is less than the {len(vocab_checks)} fields expected."
+                    )
+            if spec_verb:
+                print(errors[-1])
+            break
+        else:
+            field=vocab_checks["field"+num]
+
+            if field.startswith('__vocabs__') or field.startswith('__URL__'):
```

Comment:
Repeating the last comment on nested ifs, you can use continue to skip loops where something isn't true rather than having a nested set of conditions.